nestkernel/ntree_impl.h

```diff
@@ -478,7 +482,14 @@ Ntree< D, T, max_capacity, max_depth >::insert( Position< D > pos, const T& node
   if ( leaf_ )
   {
 
-    assert( ( pos >= lower_left_ ) && ( pos < lower_left_ + extent_ ) );
+    for ( int i = 0; i < D; ++i )
+    {
+      // Comparing against an epsilon value in case there are round-off errors.
+      // Using a negative epsilon value because the round-off error may go both ways
+      // and the difference we check against may therefore be +/- 10^-16.
+      assert( ( pos - lower_left_ )[ i ] > -std::numeric_limits< double >::epsilon()
```

Comment:
If it is because of the +/- difference, I think it is better to use `std::abs( ( pos - lower_left_ )[ i ] ) > std::numeric_limits< double >::epsilon() ... `

Do we need to do the same in the case above? Or is it only needed here?

Reply:
Using `(pos - lower_left) > -epsilon` is not the same as `abs(pos - lower_left) > epsilon`.  The first becomes `true` for `pos - lower_left` values that are higher than `-epsilon`, while the second is `true` for values higher than `+epsilon` or lower than `-epsilon`, which is not what we want. We want to check that the `pos - lower_left` value is higher than zero, but because of round-off errors it may be slightly below zero as well.

In the case above I suppose the value may also be slightly below zero too. I've added a negative sign to the epsilon there.

Reply:
Ah, very true!